src/main/kotlin/app/revanced/patches/youtube/layout/sponsorblock/bytecode/fingerprints/PlayerOverlaysLayoutInitFingerprint.kt

```diff
@@ -1,9 +1,27 @@
 package app.revanced.patches.youtube.layout.sponsorblock.bytecode.fingerprints
 
-
+import app.revanced.patcher.extensions.or
 import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
 
 object PlayerOverlaysLayoutInitFingerprint : MethodFingerprint(
-
-    customFingerprint = { methodDef -> methodDef.returnType.endsWith("YouTubePlayerOverlaysLayout;") }
+    returnType = "V",
+    access = AccessFlags.PRIVATE or AccessFlags.FINAL,
+    parameters = listOf(),
+    opcodes = listOf(
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.MOVE_RESULT_OBJECT,
+        Opcode.CHECK_CAST, // R.id.inset_overlay_view_layout
+        Opcode.IPUT_OBJECT,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.CONST,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.MOVE_RESULT_OBJECT,
+        Opcode.CHECK_CAST,
+        Opcode.NEW_INSTANCE,
+        ),
+    customFingerprint = { methodDef ->
+        methodDef.definingClass == "Lcom/google/android/apps/youtube/app/player/overlay/YouTubeControlsOverlay;"
+    }
```

Comment:
Since the fingerprint has been changed, have you considered if the existing version compatibility of this patch is still valid? If not, you can drop the old versions or test if it works.